pkg/modules/authorization.go

```diff
@@ -631,6 +633,543 @@ func AuthorizationServerDeployment(ctx context.Context, isDeleting bool, op util
+	if vaultCertificateAuthority != "" {
+		vaultCABytes, err := base64.StdEncoding.DecodeString(vaultCertificateAuthority)
+		if err != nil {
+			return fmt.Errorf("decoding vault certificate authority: %w", err)
+		}
+
+		secret := corev1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Secret",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "vault-certificate-authority",
+				Namespace: cr.Namespace,
+			},
+			Type: corev1.SecretTypeOpaque,
+			Data: map[string][]byte{
+				"ca.crt": vaultCABytes,
+			},
+		}
+
+		secretBytes, err := json.Marshal(&secret)
+		if err != nil {
+			return fmt.Errorf("marshalling vault certificate authority secret: %w", err)
+		}
+
+		yamlString, err := yaml.JSONToYAML(secretBytes)
+		if err != nil {
+			return fmt.Errorf("converting vault certificate authority json to yaml: %w", err)
+		}
+
+		err = applyDeleteObjects(ctx, ctrlClient, string(yamlString), isDeleting)
+		if err != nil {
+			return fmt.Errorf("applying vault certificate authority secret: %w", err)
+		}
+	}
+
+	// apply/delete vault-client-certificate secret if it was provided
+	if vaultCertificate != "" && vaultPrivateKey != "" {
+		vaultCertBytes, err := base64.StdEncoding.DecodeString(vaultCertificate)
+		if err != nil {
+			return fmt.Errorf("decoding vault certificate: %w", err)
+		}
+
+		vaultKeyBytes, err := base64.StdEncoding.DecodeString(vaultPrivateKey)
+		if err != nil {
+			return fmt.Errorf("decoding vault private key: %w", err)
+		}
+
+		secret := corev1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Secret",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "vault-client-certificate",
+				Namespace: cr.Namespace,
+			},
+			Type: corev1.SecretTypeTLS,
+			Data: map[string][]byte{
+				"tls.crt": vaultCertBytes,
+				"tls.key": vaultKeyBytes,
+			},
+		}
+
+		secretBytes, err := json.Marshal(&secret)
+		if err != nil {
+			return fmt.Errorf("marshalling vault certificate secret: %w", err)
+		}
+
+		yamlString, err := yaml.JSONToYAML(secretBytes)
+		if err != nil {
+			return fmt.Errorf("converting vault certificate json to yaml: %w", err)
+		}
+
+		err = applyDeleteObjects(ctx, ctrlClient, string(yamlString), isDeleting)
+		if err != nil {
+			return fmt.Errorf("applying vault certificate secret: %w", err)
+		}
+		return nil
+	}
+
+	// apply/delete storage-service-selfsigned issuer and certificate
+	issuer := createSelfSignedIssuer(cr, "storage-service-selfsigned")
+
+	issuerByes, err := json.Marshal(issuer)
+	if err != nil {
+		return fmt.Errorf("marshaling storage-service-selfsigned issuer: %v", err)
+	}
+
+	issuerYaml, err := yaml.JSONToYAML(issuerByes)
+	if err != nil {
+		return fmt.Errorf("converting storage-service-selfsigned issuer json to yaml: %v", err)
+	}
+
+	// create/delete issuer
+	err = applyDeleteObjects(ctx, ctrlClient, string(issuerYaml), isDeleting)
+	if err != nil {
+		return err
+	}
+
+	certificate := &certificate.Certificate{
```

Comment:
Can the `createSelfSignedCertificate ` function be reused here?

Reply:
Refactored to use this func.

Reply:
Manually deployed Authorization v1 and v2 to verify change.